sql/updates/world/world_2018_11_24_01.sql

```diff
@@ -0,0 +1,3348 @@
+ALTER TABLE `areatrigger_teleport`
+	ADD COLUMN `patch` TINYINT(3) UNSIGNED NOT NULL DEFAULT '0' AFTER `id`,
```

Comment:
Some tables do have both patch_min and patch_max, those are easy to understand, while some only have `patch`. I believe renaming `patch` to `patch_min` would be better for clarity in the second case.

Reply:
A different kind of patch column is needed depending on what the table stores. For entries that can be removed from the game at a later patch, like a specific spawn, or an item drop from a loot table, you need both `patch_min` and `patch_max`. In this case the data for the specific entries does not change, it is just loaded on specific patches, and the columns are not part of the primary key.

For data that is never removed from the game, but is changed at some point, you need a single `patch` column that is included in the primary key together with the entry Id. A good example of this is item prototypes (item_template). While an item can be removed from loot tables and made unavailable, Blizzard never removed the item prototypes themselves. If you had obtained it before it was made unavailable, you would still have it. Also items can change with a patch, but their entry Id remains the same, so you need the `patch` to be part of the primary key. This way when an item changes, you add a duplicate entry for that patch with the new version of the item, and the core then loads the most recent version (the entry with the highest patch that is not above the current patch).